cmd/machine-config-daemon/pivot.go

```diff
@@ -366,13 +228,26 @@ func run(_ *cobra.Command, args []string) error {
 	}
 
 	if !changed {
-		glog.Info("Already at target pivot; exiting...")
-		if exit77 {
-			os.Exit(77)
-		}
-	} else if reboot || utils.FileExists(runPivotRebootFile) {
+		glog.Info("No changes; already at target oscontainer, no kernel args provided")
+		return nil
+	}
+
+	_, err = os.Stat(runPivotRebootFile)
```

Comment:
Same question about if this should be checked when `reboot` is set via cmd line flag.

Reply:
OK, right, I thought I'd changed that but I hadn't.  Reworked it and updated that last commit again.

At a higher procedural level though this isn't net new code, we imported it from elsewhere and actually I am working on replacing it and have another PR ready to go after this one lands.  This code will mostly be dead after that in favor of reading the new encapsulated config.

Also I personally hesitate a bit around nits like this since the previous version had all passed, every push needs 3 new clusters spun up and an hour of validation...i.e. there's a velocity vs code review tradeoff.  On the other hand it is good to come to a shared team understanding of code style for future changes!

So anyways, updated :new: again!

Reply:
I tried to restrict my review to new code; I'm sorry if I missed something and re-reviewed lines. My concern here was introducing a new failure point at a place where you could be trying to force through a reboot due to some issue. That seemed like a potentially problematic state to get into.

Understood on the CI tradeoff, please keep pushing back as I come up to speed 👍